interfaces/apparmor/backend_test.go

```diff
@@ -527,6 +528,51 @@ func (s *backendSuite) TestCombineSnippets(c *C) {
 	}
 }
 
+func (s *backendSuite) TestParallelInstallCombineSnippets(c *C) {
+	restore := release.MockAppArmorLevel(release.FullAppArmor)
+	defer restore()
+	restore = apparmor.MockIsHomeUsingNFS(func() (bool, error) { return false, nil })
+	defer restore()
+	restore = apparmor.MockIsRootWritableOverlay(func() (string, error) { return "", nil })
+	defer restore()
+
+	// NOTE: replace the real template with a shorter variant
+	restoreTemplate := apparmor.MockTemplate("\n" +
+		"###VAR###\n" +
+		"###PROFILEATTACH### (attach_disconnected,mediate_deleted) {\n" +
+		"###SNIPPETS###\n" +
+		"}\n")
+	defer restoreTemplate()
+	restoreClassicTemplate := apparmor.MockClassicTemplate("\n" +
+		"#classic\n" +
+		"###VAR###\n" +
+		"###PROFILEATTACH### (attach_disconnected,mediate_deleted) {\n" +
+		"###SNIPPETS###\n" +
+		"}\n")
+	defer restoreClassicTemplate()
+	s.Iface.AppArmorPermanentSlotCallback = func(spec *apparmor.Specification, slot *snap.SlotInfo) error {
+		return nil
+	}
+	expected := `
+@{SNAP_NAME}="samba"
+@{SNAP_INSTANCE_NAME}="samba_foo"
```

Comment:
Within the context of the apparmor profile, SNAP_INSTANCE_NAME is a little weird, but I understand why you did it. For someone auditing the profile, they will wonder "what is the difference between SNAP_NAME and SNAP_INSTANCE_NAME?".

It makes sense to me for the end result to be:
```
@{SNAP_NAME}="samba_foo"
# the snap 'basename' is used to support environment settings in parallel installs
@{SNAP_BASENAME}="samba"
```

(I'm not sure if 'SNAP_BASENAME' is the best name, but something like that)

I think I prefer this because it feels like SNAP_NAME should tend to match the security label of the snap (and therefore PROFILE_DBUS). The PR would also be smaller. Granted, SNAP_NAME doesn't match the snap's environment, but in this case I don't think that matters-- it is more important that one reading or authoring the profile understands how it works and that SNAP_NAME is the normal thing we use, with SNAP_BASENAME as an exceptional case. As it stands, we are adding a new apparmor variable that is actually the normal case, with the existing SNAP_NAME as the exception, which feels wrong.

Reply:
In retrospect and talking through everything with @bboozzoo, I'm taking back this comment. Let's use SNAP_NAME and be sure to document it clearly in the profile.